app/src/crypto_helper.c

```diff
@@ -371,24 +363,68 @@ parser_error_t computeMasterFromSeed(const uint8_t seed[KEY_LENGTH],  uint8_t ma
     return parser_ok;
 }
 
+bool check_diversifier(const uint8_t d[DIVERSIFIER_LENGTH]) {
+    if(d == NULL) {
+        return parser_unexpected_error;
+    }
+
+    uint8_t hash[32] = {0};
+
+    blake2s_state state = {0};
+    blake2s_init_with_personalization(&state, 32, (const uint8_t *)KEY_DIVERSIFICATION_PERSONALIZATION, sizeof(KEY_DIVERSIFICATION_PERSONALIZATION));
+    blake2s_update(&state, (const uint8_t *)GH_FIRST_BLOCK, sizeof(GH_FIRST_BLOCK));
+    blake2s_update(&state, d, DIVERSIFIER_LENGTH);
+    blake2s_final(&state, hash, KEY_LENGTH);
+
+    return is_valid_diversifier(hash);
+}
+
 parser_error_t computeDiversifiersList(const uint8_t dk[KEY_LENGTH], uint8_t div_start_index[11], uint8_t div_l[44]) {
```

Comment:
nit: You can define a diversifier type as uint8 div[11] and you might want to use an array of that type instead of a plain 44 for the size. This is just a comment in case that you find it useful!